src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -976,4 +976,89 @@
       it { expect(response).to redirect_to(project_show_path(apache_maintenance_incident_project)) }
     end
   end
+
+  describe 'POST #move_path' do
+    context 'valid project' do
+      let(:repository) { create(:repository, project: apache_project) }
+      let(:path_element) { create(:path_element, repository: repository) }
+
+      context 'without direction' do
+        it 'raises an exception' do
+          expect { post :move_path, params: { project: apache_project } }.to raise_error ActionController::ParameterMissing
+        end
+      end
+
+      context 'only one path_element' do
+        let(:position) { path_element.position}
+
+        context 'direction up' do
+          before do
+            post :move_path, params: { project: apache_project, repository: repository.id, direction: 'up', path: path_element }
+          end
+
+          it 'goes up' do
+            expect(path_element.reload.position).to eq(position)
+            expect(flash[:notice]).to eq("Path moved up successfully")
+            expect(response).to redirect_to({ action: :index, controller: :repositories, project: apache_project })
+          end
+        end
+
+        context 'direction down' do
+          before do
+            post :move_path, params: { project: apache_project, repository: repository.id, direction: 'down', path: path_element }
+          end
+
+          it 'goes down' do
+            expect(path_element.reload.position).to eq(position)
+            expect(flash[:notice]).to eq("Path moved down successfully")
+            expect(response).to redirect_to({ action: :index, controller: :repositories, project: apache_project })
+          end
+        end
+      end
+
+      context 'three path elements' do
+        let(:path_element_2) { create(:path_element, repository: repository) }
+        let(:path_element_3) { create(:path_element, repository: repository) }
+        let(:path_elements) { [path_element, path_element_2, path_element_3] }
```

Comment:
> > > To evaluate the three elements at a time only using path_elements
> 
> > but you are not doing that, right?
> 
> It is done in lines 1023, 1032, 1045 and 1054.

No... You are only evaluating one of them in each line and each time is a different tests... :confused: 


> > You can also reload only what you need. I'll write this down 😉
> 
> It can be reloaded only what it is expected to change, and I understand your approach, but I prefer to check that nothing has changed, reloading all the `path_elements` array.

The thing is that you reload the whole array but then you only check that one thing has not changed...